dtfs-central-api/src/repositories/tfm-facilities-repo/tfm-facility.schema.ts

```diff
@@ -0,0 +1,36 @@
+import z from 'zod';
+import { MongoObjectIdSchema } from '../../v1/routes/middleware/payload-validation/schemas';
+
+const TfmFacilityDateFieldSchema = z
+  .object({
+    $date: z.coerce.date(),
```

Comment:
z.coerce.date() just calls `new Date` (if i recall) so does weird stuff like parsing `'1'` as `Mon Jan 01 2001`. Would you be happy to handroll something using `parseISO` and `toDate` from date-fns?

Reply:
It does just call `new Date`. I am happy to leave this as a standard `z.coerce.date()` because this is something that should be validated on the way in rather than the way out